tests/manage/test_dynamic_pvc_accessmodes_with_reclaim_policies.py

```diff
@@ -0,0 +1,375 @@
+        logger.info(f"Creating two pods using same PVC {pvc_obj.name}")
+
+        # Create first pod
+        if interface_type == constants.CEPHBLOCKPOOL:
+            pod_data = templating.load_yaml_to_dict(constants.CSI_RBD_POD_YAML)
+        elif interface_type == constants.CEPHFILESYSTEM:
+            pod_data = templating.load_yaml_to_dict(
+                constants.CSI_CEPHFS_POD_YAML
+            )
+
+        pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pod'
+        )
+        pod_data['metadata']['namespace'] = self.namespace
+        pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = (
+            pvc_obj.name
+        )
+        pod_obj = helpers.create_pod(**pod_data)
+
+        node_pod1 = pod_obj.get()['spec']['nodeName']
+
+        # Create second pod
+        # Try creating pod until it is on a different node than first pod
+        for retry in range(1, 6):
+            pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+                'test', 'pod'
+            )
+            pod_obj2 = helpers.create_pod(wait=False, **pod_data)
+            node_pod2 = pod_obj2.get()['spec']['nodeName']
+            if node_pod1 != node_pod2:
+                break
+            logger.info(
+                f"Both pods are on same node. Deleting second pod and "
+                f"creating another pod. Retry count:{retry}"
+            )
+            pod_obj2.delete()
+            if retry == 5:
+                raise UnexpectedBehaviour(
+                    "Second pod is always created on same node as of first "
+                    "pod even after trying 5 times."
+                )
+
+        # Run IO on first pod
+        logger.info(f"Running IO on first pod {pod_obj.name}")
+        pod_obj.run_io('fs', '512M')
+        logging.info(f"Waiting for IO results from pod {pod_obj.name}")
+        fio_result = pod_obj.get_fio_results()
+        logging.info("IOPs after FIO:")
+        logging.info(
+            f"Read: {fio_result.get('jobs')[0].get('read').get('iops')}"
+        )
+        logging.info(
+            f"Write: {fio_result.get('jobs')[0].get('write').get('iops')}"
+        )
+
+        # Fetch usage details
+        mount_point = pod_obj.exec_cmd_on_pod(command="df -kh")
+        mount_point = mount_point.split()
+        usage = mount_point[mount_point.index('/var/lib/www/html') - 1]
+
+        # Verify that second pod is not in Running state
+        try:
+            assert not pod_obj2.ocp.wait_for_resource(
+                condition='Running', resource_name=pod_obj2.name,
+            ), "Unexpected: Second pod is in Running state"
+        except TimeoutExpiredError:
+            logger.info(
+                f"Verified: Second pod {pod_obj2.name} is not in "
+                f"Running state"
+            )
+
+        # Delete first pod
+        pod_obj.delete(wait=True)
+
+        # Verify pod is deleted
```

Comment:
It's better to have this checked as part of the 'pod.delete()'. that way we don't have to repeat the code to verify deletion in all the scripts we write. I know this is part of upper classes implementation. but just a suggestion...

Reply:
@hnallurv the delete() function is generic for all KINDs, hence if we want this check in delete function,  we would need to perform different checks for different kinds.  

Please suggest if my understanding is incorrect

Reply:
Now using wait_for_delete()